main.tf

```
@@ -1,22 +1,17 @@
-# ARM template for Service Bus subscription
-data "template_file" "subscription_template" {
-  template = "${file("${path.module}/template/subscription_template.json")}"
-}
+resource "azurerm_servicebus_subscription" "servicebus_subscription" {
+  name                = var.name
+  resource_group_name = var.resource_group_name
+  namespace_name      = var.namespace_name
+  topic_name          = var.topic_name
 
-# Create Azure Service Bus subscription
-resource "azurerm_template_deployment" "subscription" {
-  template_body       = "${data.template_file.subscription_template.rendered}"
-  name                = "${var.name}"
-  deployment_mode     = "Incremental"
-  resource_group_name = "${var.resource_group_name}"
+  lock_duration                     = var.lock_duration
+  max_delivery_count                = var.max_delivery_count
+  forward_to                        = var.forward_to
+  forward_dead_lettered_messages_to = var.forward_dead_lettered_messages_to
 
-  parameters = {
-    serviceBusNamespaceName       = "${var.namespace_name}"
-    serviceBusTopicName           = "${var.topic_name}"
-    serviceBusSubscriptionName    = "${var.name}"
-    lockDuration                  = "${var.lock_duration}"
-    maxDeliveryCount              = "${var.max_delivery_count}"
-    forwardTo                     = "${var.forward_to}"
-    forwardDeadLetteredMessagesTo = "${var.forward_dead_lettered_messages_to}"
-  }
+  requires_session                     = false
```

you created a variable for this but aren't using it
```suggestion
  requires_session                     = var.requires_session
```